db/active_replicator_pull.go

```diff
@@ -148,38 +179,55 @@ func (apr *ActivePullReplicator) Complete() {
 }
 
 func (apr *ActivePullReplicator) _initCheckpointer() error {
+	// wrap the replicator context with a cancelFunc that can be called to abort the checkpointer from _disconnect
+	apr.checkpointerCtx, apr.checkpointerCtxCancel = context.WithCancel(apr.ctx)
 
 	checkpointHash, hashErr := apr.config.CheckpointHash()
 	if hashErr != nil {
 		return hashErr
 	}
 
-	apr.Checkpointer = NewCheckpointer(apr.checkpointerCtx, apr.CheckpointID, checkpointHash, apr.blipSender, apr.config, apr.getPullStatus)
+	err := apr.forEachCollection(func(c *activeReplicatorCollection) error {
+		c.Checkpointer = NewCheckpointer(apr.checkpointerCtx, c.dataStore, apr.CheckpointID, checkpointHash, apr.blipSender, apr.config, apr.getPullStatus, c.collectionIdx)
 
-	var err error
-	apr.initialStatus, err = apr.Checkpointer.fetchCheckpoints()
-	base.InfofCtx(apr.ctx, base.KeyReplicate, "Initialized pull replication status: %+v", apr.initialStatus)
+		if !apr.config.CollectionsEnabled {
+			err := c.Checkpointer.fetchCheckpoints()
```

Comment:
Since the checkpointer is scoped to the datastore, why do we not need to fetch checkpoints for other collections?

I think this is because we get the checkpoints from `GetCollections` message, but then maybe `fetchCheckpoint` should be renamed to something that indicates it is for the default collection only.

Reply:
Yep, fair point - renamed this to be default collection-specific